### ISSUE TITLE
React with typescript - build fails for a type mismatch in common/main.d.ts 

### ISSUE THREAD

**Issue by cfall** [fullcalendar/fullcalendar-react#206]
### Bug Reports
We integrated fullcalendar-react in our react app. The integration and tests during development were smooth but when we tried to deploy the app in our QA environment the build failed for a type mismatch error in the file common/main.d.ts

Here's the full stack trace of the error: 

node_modules/@fullcalendar/common/main.d.ts:2761:5 - error TS2416: Property 'render' in type 'NowTimer' is not assignable to the same property in base type 'Component<NowTimerProps, NowTimerState, any>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2761     render(): ComponentChildren;
         ~~~~~~

node_modules/@fullcalendar/common/main.d.ts:2761:5 - error TS2416: Property 'render' in type 'NowTimer' is not assignable to the same property in base type 'Component<NowTimerProps, NowTimerState, any>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2761     render(): ComponentChildren;
         ~~~~~~

node_modules/@fullcalendar/common/main.d.ts:2761:5 - error TS2416: Property 'render' in type 'NowTimer' is not assignable to the same property in base type 'Component<NowTimerProps, NowTimerState, any>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2761     render(): ComponentChildren;
         ~~~~~~

node_modules/@fullcalendar/common/main.d.ts:2761:5 - error TS2416: Property 'render' in type 'NowTimer' is not assignable to the same property in base type 'Component<NowTimerProps, NowTimerState, any>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2761     render(): ComponentChildren;
         ~~~~~~

render() function of the base class is expected to return ReactNode while in the child classes it's ComponentChildren.

**Comment by ewillroth** [comment]
Seeing this same error with some additional types mentioned. I am using typescript @ ^4.8.2

node_modules/@fullcalendar/common/main.d.ts:426:5 - error TS2416: Property 'render' in type 'MountHook<ContentArg>' is not assignable to the same property in base type 'BaseComponent<MountHookProps<ContentArg>, Dictionary>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.
      Type 'object' is not assignable to type 'ReactNode'.

426     render(): ComponentChildren;
        ~~~~~~

node_modules/@fullcalendar/common/main.d.ts:2431:5 - error TS2416: Property 'render' in type 'CalendarDataProvider' is not assignable to the same property in base type 'Component<CalendarDataProviderProps, CalendarData, any>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2431     render(): ComponentChildren;
         ~~~~~~

node_modules/@fullcalendar/common/main.d.ts:2522:5 - error TS2416: Property 'render' in type 'CalendarRoot' is not assignable to the same property in base type 'BaseComponent<CalendarRootProps, CalendarRootState>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2522     render(): ComponentChildren;
         ~~~~~~

node_modules/@fullcalendar/common/main.d.ts:2761:5 - error TS2416: Property 'render' in type 'NowTimer' is not assignable to the same property in base type 'Component<NowTimerProps, NowTimerState, any>'.
  Type '() => ComponentChildren' is not assignable to type '() => ReactNode'.
    Type 'ComponentChildren' is not assignable to type 'ReactNode'.

2761     render(): ComponentChildren;
         ~~~~~~

**Comment by ewillroth** [comment]
I've created a reduced test case in the repo here: https://github.com/ewillroth/fullCalendarExample

I was unable to use code sandbox since the issue occurs on build.

**Comment by acerix** [comment]
You can use the example project as a starting point:

https://github.com/fullcalendar/fullcalendar-example-projects/tree/master/react-typescript

If there is a bug, please supply a runnable demo,  Codesandbox will show any build errors.

**Comment by ewillroth** [comment]
Here is a runnable demo:

https://codesandbox.io/p/sandbox/late-https-b0x7j1

**Comment by ewillroth** [comment]
Just to follow up for @cfall (in case you are still having this issue) - I was able to fix this by updating the `compilerOptions.skipLibCheck` in my `tsonfig.json`. 
https://www.typescriptlang.org/tsconfig#skipLibCheck

**Comment by Amwam** [comment]
I think this might be the issue fixed in #201 

Can see in the lockfile here, that there are two versions of the @fullcalendar/common library, causing conflicts
https://github.com/ewillroth/fullCalendarExample/blob/main/yarn.lock#L1181

**Comment by cfall** [comment]
> Just to follow up for @cfall (in case you are still having this issue) - I was able to get around this error by updating the `compilerOptions.skipLibCheck` in my `tsonfig.json`. Essentially turning off type checking for `node_modules` to ignore the mismatched types.
> 
> https://www.typescriptlang.org/tsconfig#skipLibCheck

Thank you @ewillroth I used it as a temporary fix

**Comment by arshaw** [comment]
The PR has been merged and the fix for this has been released in [v5.11.3](https://github.com/fullcalendar/fullcalendar-react/releases/tag/v5.11.3). Sorry for the trouble folks.

**Comment by cfall** [comment]
> The PR has been merged and the fix for this has been released in [v5.11.3](https://github.com/fullcalendar/fullcalendar-react/releases/tag/v5.11.3). Sorry for the trouble folks.

Thank you guys @arshaw for the great work and this amazing lib
